standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -281,6 +279,7 @@ public static boolean isRenameAllowed(Database srcDB, Database destDB) {
     static AtomicInteger databaseCount, tableCount, partCount;
 
     private Warehouse wh; // hdfs warehouse
+    private static Striped<Lock> tablelocks;
```

Comment:
i think, it doesn't have to be static

Reply:
this field must be static - if it's not; then it could not impose exclusiveness.

note that this field is [initialized alongside with the `threadPool` field](https://github.com/apache/hive/pull/1724/files/bc11bb62d3700b227f48b4683282b826b946f8bd#diff-00e70b6958060aa36762b21bf16676f83af01c1e09b56816aecc6abe7c8ac866R491)

Reply:
oh, sorry, i missed that synchronization is done on class level not instance. 
Note: there is a bug in synchronization, double checked locking should be used here or if check should be moved inside the critical section.